### PR TITLE
Issue 6874: Flaky test SecureStreamMetadataTasksTest.readerGroupsTest is failing

### LINKED ISSUE #6874: Flaky test SecureStreamMetadataTasksTest.readerGroupsTest is failing
**Describe the bug**
Unit test io.pravega.controller.task.Stream.SecureStreamMetadataTasksTest.readerGroupsTest is failing with below exception in jenkins job. This never fails in local. 

java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at io.pravega.controller.task.Stream.StreamMetadataTasksTest.readerGroupsTest(StreamMetadataTasksTest.java:720)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)

### PR DESCRIPTION
**Pull request by munish1789**
**Change log description**  
Fix flaky test SecureStreamMetadataTasksTest.readerGroupsTest 

**Purpose of the change**  
Fixes #6874 

**What the code does**  
Asserting using assertEventuallyEquals instead of assertTrue 

**How to verify it**  
Jenkins job should not have SecureStreamMetadataTasksTest failure

### CODE REVIEW

File: controller/src/test/java/io/pravega/controller/server/eventProcessor/RequestHandlersTest.java [hunk 1]
```diff
@@ -840,6 +841,7 @@ public long getRequestId() {
         future.join();
     }
 
+    @SneakyThrows
```
Comment (RaulGracia) [inline controller/src/test/java/io/pravega/controller/server/eventProcessor/RequestHandlersTest.java:844]: Why do you need this? You could add the exception in the test signature itself.

File: controller/src/test/java/io/pravega/controller/server/eventProcessor/ScaleRequestHandlerTest.java [hunk 2]
```diff
@@ -205,6 +206,7 @@ public void tearDown() throws Exception {
         }
     }
 
+    @SneakyThrows
```
Comment (RaulGracia) [inline controller/src/test/java/io/pravega/controller/server/eventProcessor/ScaleRequestHandlerTest.java:209]: Same here.

File: controller/src/test/java/io/pravega/controller/task/KeyValueTable/TableMetadataTasksTest.java [hunk 3]
```diff
@@ -121,6 +122,7 @@ public void tearDown() throws Exception {
         }
     }
 
+    @SneakyThrows
```
Comment (RaulGracia) [inline controller/src/test/java/io/pravega/controller/task/KeyValueTable/TableMetadataTasksTest.java:125]: Same here.

File: controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java [hunk 4]
```diff
@@ -483,6 +484,7 @@ public void updateStreamSegmentCountScalingPolicyTest() throws Exception {
                 etr.getObject().getActiveEpoch(), 0L).join().getStatus(), Controller.ScaleStatusResponse.ScaleStatus.SUCCESS);
     }
 
+    @SneakyThrows
```
Comment (RaulGracia) [inline controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java:487]: Same here.

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Looks good, just one minor comment. Please, test this in Jenkins several times to see if it actually solves the issue.

File: controller/src/test/java/io/pravega/controller/server/eventProcessor/RequestHandlersTest.java (hunk 1 above)
Comment (munish1789) [reply controller/src/test/java/io/pravega/controller/server/eventProcessor/RequestHandlersTest.java]: used throws Exception

**Comment by RaulGracia** [comment]
Found another occurrence of:
```
io.pravega.controller.store.stream.ZKCounterTest > testCounter FAILED
    java.util.concurrent.CompletionException: java.lang.StackOverflowError
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
        at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1113)
        at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2235)
```
which is unrelated. Triggering again.

**Review by RaulGracia — CHANGES_REQUESTED** [review]
There is at least one more change needed: in `readerGroupsTest()`, please join the future returned from `streamMetadataTasks.updateReaderGroup(SCOPE, "rg2", newConfig, 0L):` (line 615)

**Comment by munish1789** [comment]
it is getting joined in line no 617

 CompletableFuture<Controller.UpdateReaderGroupResponse> updateResponse =
        streamMetadataTasks.updateReaderGroup(SCOPE, "rg2", newConfig, 0L);
        assertTrue(Futures.await(processEvent(requestEventWriter)));
        Controller.UpdateReaderGroupResponse updateResponseResult = updateResponse.join();

**Comment by RaulGracia** [comment]
You are right @munish1789 the indentation confused me. If you could add spaces to the broken line, it would help.

**Review by ShwethaSNayak — COMMENTED** [review]
LGTM

**Review by RaulGracia — CHANGES_REQUESTED** [review]
We got another occurrence of the error even with these changes, so we are missing something else.

**Comment by RaulGracia** [comment]
This has been fixed already in #6887.

**Pull request closed without merging**